### PR TITLE
Network improve and fix wireless

### PR DESCRIPTION
**Pull request by rvykydal**

### CODE REVIEW

**Review by M4rtinK — APPROVED** [review]
Looks good to me, nice cleanup! :)

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: pyanaconda/ui/gui/spokes/network.py [hunk 1]
```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
 secret_agent = None
 
 
-class NotAuthorizedException(dbus.DBusException):
-    _dbus_error_name = SECRET_AGENT_IFACE + '.NotAuthorized'
-
+class SecretAgent(object):
+    __dbus_xml__ = """
+    <node>
+        <interface name="org.freedesktop.NetworkManager.SecretAgent">
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:1293]: I would consider to use type hints and generate the DBus API. You can check that that the expected specification is the same as the generated one in a unit test.

File: pyanaconda/ui/gui/spokes/network.py [hunk 2]
```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
...
                 if secret['key']:
-                    secrets[secret['key']] = secret['value']
+                    secrets[secret['key']] = Variant('s', secret['value'])
 
-        settings = dbus.Dictionary({setting_name: secrets})
+        settings = {setting_name: secrets}
 
         return settings
 
+    def _unpack_connection_hash(self, connection_hash):
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:1336]: Maybe you can use `get_native` or `unwrap_variant` from `dasbus.typing`.

File: pyanaconda/ui/gui/spokes/network.py [hunk 3]
```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
...
         with self.spoke.main_window.enlightbox(dialog.window):
             rc = dialog.run()
 
-        secrets = dbus.Dictionary()
+        secrets = dict()
         if rc == 1:
             for secret in content['secrets']:
                 if secret['key']:
-                    secrets[secret['key']] = secret['value']
+                    secrets[secret['key']] = Variant('s', secret['value'])
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:1330]: Please, use `get_variant` from `dasbus.typing`.

File: pyanaconda/ui/gui/spokes/network.py [hunk 4]
```diff
@@ -1415,19 +1425,21 @@ def _validate_staticwep(self, secret):
...
 
     global secret_agent
     if not secret_agent:
-        # Ignore an error from pylint incorrectly analyzing types in dbus-python
-        secret_agent = SecretAgent(spoke) # pylint: disable=no-value-for-parameter
-        bus = dbus.SystemBus()
-        proxy = bus.get_object(NM_SERVICE, AGENT_MANAGER_PATH)
-        proxy.Register("anaconda", dbus_interface=AGENT_MANAGER_IFACE)
+        secret_agent = SecretAgent(spoke)
+        bus = SystemMessageBus()
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:1441]: Please, use `SystemBus` from `pyanaconda.core.dbus`.

File: pyanaconda/ui/gui/spokes/network.py [hunk 5]
```diff
@@ -41,35 +42,30 @@
...
 NM._80211ApFlags = getattr(NM, "80211ApFlags")
 NM._80211ApSecurityFlags = getattr(NM, "80211ApSecurityFlags")
 NM._80211Mode = getattr(NM, "80211Mode")
 
-NM_SERVICE = "org.freedesktop.NetworkManager"
 NM_SECRET_AGENT_GET_SECRETS_FLAG_ALLOW_INTERACTION = 0x1
-SECRET_AGENT_IFACE = 'org.freedesktop.NetworkManager.SecretAgent'
-AGENT_MANAGER_IFACE = 'org.freedesktop.NetworkManager.AgentManager'
 AGENT_MANAGER_PATH = "/org/freedesktop/NetworkManager/AgentManager"
+SECRET_AGENT_PATH = "/org/freedesktop/NetworkManager/SecretAgent"
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:68]: You can also create instances of `DBusObjectIdentifier` for the agent manager and the secret agent.

File: pyanaconda/ui/gui/spokes/network.py [hunk 6]
```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
 secret_agent = None
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:1287]: Maybe I would move the code for the secret agent to a new file.

File: pyanaconda/ui/gui/spokes/network.py [hunk 7]
```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
...
                   connection_path, setting_name, str(hints), flags)
         if not (flags & NM_SECRET_AGENT_GET_SECRETS_FLAG_ALLOW_INTERACTION):
             return
 
-        content = self._get_content(setting_name, connection_hash)
+        unpacked_connection_hash = self._unpack_connection_hash(connection_hash)
+
+        content = self._get_content(setting_name, unpacked_connection_hash)
         dialog = SecretAgentDialog(self.spoke.data, content=content)
         with self.spoke.main_window.enlightbox(dialog.window):
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/network.py:1323]: If you replace the spoke with a callback, you might be able to write unit tests for this class.

**Review by poncovka — COMMENTED** [review]
I think that it might be possible to write unit tests for the secret agent.

**Comment by rvykydal** [comment]
I am going to address fixes and improvements more thoroughly in https://github.com/rhinstaller/anaconda/pull/2977
and migration to dasbus in a sparate PR, hopefully addressing Vendy's review.

**Comment by rvykydal** [comment]
> I am going to address fixes and improvements more thoroughly in #2977
> and migration to dasbus in a sparate PR, hopefully addressing Vendy's review.

The new PR: https://github.com/rhinstaller/anaconda/pull/2995

**Pull request closed without merging**